include/render.hpp

```diff
@@ -146,15 +146,21 @@ void render(sycl::queue& queue, std::array<color, width * height>& fb,
       sycl::buffer<color, 2>(fb.data(), sycl::range<2>(height, width));
   auto hittables_buf = sycl::buffer<hittable_t, 1>(hittables.data(),
                                                    sycl::range<1>(nb_hittable));
+  auto cam_buf = sycl::buffer<camera, 1>(&cam, sycl::range<1>(1));
 
   // Submit command group on device
-  queue.submit([=, &hittables_buf, &frame_buf, &cam](sycl::handler& cgh) {
+  queue.submit([=, &hittables_buf, &frame_buf, &cam_buf](sycl::handler& cgh) {
```

Comment:
Keep your life easy: remove all the explicit captures...
Also `queue.submit` accepts a lambda using `[&]`